### PR TITLE
Add native PGP keyserver lookups (Issue 2159).

### PR DESCRIPTION
**Pull request by philratcliffe**
Updates so that Mailpile performs PGP key server lookups directly rather than through GnuPG.

### CODE REVIEW

**Comment by BjarniRunar** [comment]
Thanks for this, and apologies for the lack of reactions. I've been going through a bit of burnout lately, so I've been spending time on other things. I'll take a look as soon as I can.

**Comment by philratcliffe** [comment]
No worries @BjarniRunar; thanks for creating such a great email client!

**Comment by fiza1160** [comment]
#2159

File: mailpile/plugins/keylookup/__init__.py [hunk 1]
```diff
@@ -513,51 +518,92 @@ class KeyserverLookupHandler(LookupHandler):
...
+    PRIVACY_FRIENDLY = True
     PRIORITY = 200
     SCORE = 1
 
+    # During testing, there were frequent HTTP gateway errors returned from
+    # hkps.pool.sks-keyservers.net so sks-keyservers.net was added too.
+    KEY_SERVER_BASE_URLS = [
+        "https://sks-keyservers.net/pks/lookup",
+        "https://hkps.pool.sks-keyservers.net/pks/lookup",
+    ]
```
Comment (wiktor-k) [inline mailpile/plugins/keylookup/__init__.py]: In my experience https://keyserver.ubuntu.com/ has the highest uptime and lowest response times.

**Comment by BjarniRunar** [comment]
Just a note: Merging this has become higher priority for me, since it will allow Mailpile to put upper limits on key size and thus protect GnuPG from SKS keyserver spam, as described in https://gist.github.com/rjhansen/67ab921ffb4084c865b3618d6955275f

For now we've reconfigured Mailpile in master/nightly to avoid the SKS keyservers entirely, but I think it would still be valuable to our users to be able to fall back to the keyserver pool as a last resort, simply because all the modern replacements are still at an early stage of development.

I'm going to try and find time to look at this, this week. Sorry again about the long delay.

**Pull request merged**